.setup/bin/setup_sample_courses.py

```diff
@@ -980,6 +1019,47 @@ def create(self):
                                         src = os.path.join(gradeable.lichen_sample_path, gradeable.plagiarism_submissions.pop())
                                         dst = os.path.join(submission_path, str(version))
                                         create_gradeable_submission(src, dst)
+                                elif gradeable.annotated_pdf is True:
+                                    if version == versions_to_submit:
+                                        annotation_version_path = os.path.join(annotation_path, str(versions_to_submit))
+                                        if not os.path.exists(annotation_version_path):
+                                            os.makedirs(annotation_version_path)
+                                            os.system("chown -R submitty_php:{}_tas_www {}".format(self.code, annotation_version_path))
+                                    
+                                    # Get a list of graders that has access to the submission
+                                    assigned_graders = []
+                                    stmt = select([
+                                        peer_assign.columns.user_id,
+                                        peer_assign.columns.grader_id
+                                    ]).where(
+                                        peer_assign.columns.user_id == user.id
+                                    )
+                                    for res in self.conn.execute(stmt):  
+                                        assigned_graders.append(res[1])
+
+                                    submissions = random.sample(gradeable.submissions, random.randint(1, len(gradeable.submissions)))
+                                    for submission in submissions:
+                                        src = os.path.join(gradeable.sample_path, submission)
+                                        dst = os.path.join(submission_path, str(version))
+                                        create_gradeable_submission(src, dst)
+
+                                        if version == versions_to_submit:
+                                            annotations = random.sample(gradeable.annotations, random.randint(1, len(gradeable.annotations)))
+                                            graders = random.sample(assigned_graders, len(annotations)-1)
+                                            # Make sure instructor is responsible for one of the annotations
+                                            graders.append("instructor")
+
+                                            anon_dst = os.path.join(dst, submission).split("/")
+                                            anon_dst[9] = user.anon_id
+                                            anon_dst = "/".join(anon_dst) # has the user id in the file path being anonymous
```

Comment:
Is there a better way to write this?

Reply:
I followed the same logic as `getAnonPath()` in other PHP files so I'm not sure if there is a better way.

Reply:
Part of the point of using `os.path.join()` is that this could theoretically be run on systems which have separators other than `/`.  Thus, it is a little odd to split the path on slashes after using a function to join the path for you.  I suppose this works well enough in the default development environment, so is probably not worth the time to change.